src/oldreader_api.cpp

```diff
@@ -66,24 +66,26 @@ std::string oldreader_api::retrieve_auth() {
 	}
 
 	std::string pass = cfg->get_configvalue("oldreader-password");
-    std::string pass_file = cfg->get_configvalue("oldreader-passwordfile");
-    std::string pass_eval = cfg->get_configvalue("oldreader-passwordeval");
+	std::string pass_file = cfg->get_configvalue("oldreader-passwordfile");
+	std::string pass_eval = cfg->get_configvalue("oldreader-passwordeval");
 	if ( pass == "" ) {
-        if ( pass_file != "" ) {
-            pass = read_password(pass_file.c_str());
-        } else if ( pass_eval != "") {
-            pass = eval_password(pass_eval.c_str());
-        }
+		if ( pass_file != "" ) {
+			pass = read_password(pass_file);
+			if ( pass == "" ) return "";
+		} else if ( pass_eval != "") {
+			pass = eval_password(pass_eval);
+			if ( pass == "" ) return "";
```

Comment:
> Maybe it would make more sense to have it read password and if this is empty check either passwordfile, passwordeval or read from cin if neither is set and if this didn't work return (since it doesn't make sense to authenticate with an empty string as password).

I concur. Make it a separate commit, though—mixing refactorings with actual changes to logic is *bad*. Thanks!